src/main/java/org/jenkinsci/plugins/workflow/cps/Snippetizer.java

```diff
@@ -349,17 +350,69 @@ private static boolean isSingleMap(Map<String, ?> args) {
         return Jenkins.getActiveInstance().getDescriptorByName(id);
     }
 
-    @Restricted(DoNotUse.class)
-    public Collection<? extends StepDescriptor> getStepDescriptors(boolean advanced) {
-        TreeSet<StepDescriptor> t = new TreeSet<StepDescriptor>(new StepDescriptorComparator());
+    @Restricted(NoExternalUse.class)
+    public Collection<QuasiDescriptor> getQuasiDescriptors(boolean advanced) {
+        TreeSet<QuasiDescriptor> t = new TreeSet<>();
         for (StepDescriptor d : StepDescriptor.all()) {
             if (d.isAdvanced() == advanced) {
-                t.add(d);
+                t.add(new QuasiDescriptor(d));
+                if (d.isMetaStep()) {
+                    DescribableModel<?> m = new DescribableModel<>(d.clazz);
+                    Collection<DescribableParameter> parameters = m.getParameters();
+                    if (parameters.size() == 1) {
+                        DescribableParameter delegate = parameters.iterator().next();
+                        if (delegate.isRequired()) {
+                            if (delegate.getType() instanceof HeterogeneousObjectType) {
+                                // TODO HeterogeneousObjectType does not yet expose symbol information, and DescribableModel.symbolOf is private
+                                for (DescribableModel<?> delegateOptionSchema : ((HeterogeneousObjectType) delegate.getType()).getTypes().values()) {
+                                    Class<?> delegateOptionType = delegateOptionSchema.getType();
+                                    Descriptor<?> delegateDescriptor = Jenkins.getActiveInstance().getDescriptorOrDie(delegateOptionType.asSubclass(Describable.class));
+                                    Symbol symbol = delegateDescriptor.getClass().getAnnotation(Symbol.class);
+                                    if (symbol != null && symbol.value().length > 0) {
+                                        t.add(new QuasiDescriptor(delegateDescriptor));
+                                    }
+                                }
+                            }
+                        }
+                    } // TODO currently not handling metasteps with other parameters, either required or (like GenericSCMStep) not
+                }
             }
         }
         return t;
     }
 
+    /** Similar to {@link StepDescriptor} but could also represent a metastep’s delegate. */
```

Comment:
IIUC, I suggest this javadoc to be expanded to something like this:

---

{@code QuasiDescriptor} represents a step and other step-like things that show up in Snippetizer, which can generate some Pipeline Script fragment.

{@link #real} can be a {@code StepDescriptor}, in which case Snippetizer generates an invocation of that step. Or it can be a {@link Descriptor} that can be executed via {@linkplain StepDescriptor#isMetaStep() meta step}, because from users' point of view a regular Describable (say {@code SimpleBuildStep}) executed via a meta step is indistinguishable from a {@code Step}.